pkg/ipamd/datastore/data_store.go

```diff
@@ -1465,22 +1471,27 @@ func (ds *DataStore) CheckFreeableENIexists() bool {
 }
 
 // NormalizeCheckpointDataByPodVethExistence will normalize checkpoint data by removing allocations that do not have a corresponding pod veth.
-// This should not happen unless container runtimes have bugs that failed to invoke the cmdDel for died pods.
-// We use this reconciliation as a safety mechanism during transition from CRI to state file.
+// This can happen if pods are deleted while IPAMD is inactive.
 func (ds *DataStore) normalizeCheckpointDataByPodVethExistence(checkpoint CheckpointData) (CheckpointData, error) {
 	hostNSLinks, err := ds.netLink.LinkList()
 	if err != nil {
 		return CheckpointData{}, err
 	}
 	var validatedAllocations []CheckpointEntry
+	var staleAllocations []CheckpointEntry
 	for _, allocation := range checkpoint.Allocations {
 		if err := ds.validateAllocationByPodVethExistence(allocation, hostNSLinks); err != nil {
-			ds.log.Warnf("ignore IP allocation for %v:%v,%v due to %v", allocation.ContainerID, allocation.IPv4, allocation.IPv6, err)
+			ds.log.Warnf("stale IP allocation for ID(%v): IPv4(%v), IPv6(%v) due to %v", allocation.ContainerID, allocation.IPv4, allocation.IPv6, err)
+			staleAllocations = append(staleAllocations, allocation)
```

Comment:
Right, but we can only do that if the pod was created by v1.12.1+. It would be great to remove this overhead in the future, but in the common case, there should never be any stale allocations to clean up